src/main/java/net/neoforged/neoforge/event/AddPackFindersEvent.java

```diff
@@ -44,4 +53,28 @@ public void addRepositorySource(RepositorySource source) {
     public PackType getPackType() {
         return packType;
     }
+
+    /**
+     * Helper method to register a pack found under the `resources/` folder.
+     *
+     * @param packLocation    Location of the pack to load. Namespace should be the modid of the pack owner and path is the location under `resources/` folder
+     * @param packType        Whether pack is a resourcepack or datapack
+     * @param packNameDisplay The text that shows for the pack on the pack selection screen
+     * @param packSource      Controls whether the datapack is enabled or disabled by default. Resourcepacks are always disabled by default unless you set alwaysActive to true.
+     * @param alwaysActive    Whether the pack is forced active always. If false, players have to manually activate the pack themselves
+     * @param packPosition    Where the pack goes for determining pack applying order
+     */
+    public void addPackFinders(ResourceLocation packLocation, PackType packType, Component packNameDisplay, PackSource packSource, boolean alwaysActive, Pack.Position packPosition) {
+        if (getPackType() == packType) {
+            var resourcePath = ModList.get().getModFileById(packLocation.getNamespace()).getFile().findResource(packLocation.getPath());
+
+            var pack = Pack.readMetaAndCreate(
+                    new PackLocationInfo(packLocation.toString(), packNameDisplay, packSource, Optional.empty()),
```

Comment:
You're using the pack location, a resource location, as the pack ID here -- this is probably a bad idea, as the format will be `namespace:path` while the vanilla format is `type/info`. Maybe do `"builtin/"+packLocation.toString()`?

Reply:
Since one can pick a different pack source besides builtin/, won’t that make the builtin/ not true anymore? Or is that meaning the mod is providing the pack

Reply:
Just pick some relevant prefix, it doesn't really have to be the pack source in the context of modded stuff so long as the convention is followed. For instance, mod-included packs by default are `mod/modid`; could even do `mod/namespace:path` here if you wanted, since it's coming from the mod jar.

Reply:
https://github.com/neoforged/NeoForge/pull/928/commits/cdb2d2694b766e74649339da3c6c1cf73162f21c